lib/number_format.js

```diff
@@ -31,10 +31,14 @@ var propTypes = {
   format: _react.PropTypes.oneOfType([_react.PropTypes.string, _react.PropTypes.func]),
   mask: _react.PropTypes.string,
   value: _react.PropTypes.oneOfType([_react.PropTypes.number, _react.PropTypes.string]),
-  customInput: _react.PropTypes.func
+  customInput: _react.PropTypes.func,
+  allowNegative: _react.PropTypes.bool,
+  onKeyDown: _react.PropTypes.func,
+  onChange: _react.PropTypes.func
 };
 
 var defaultProps = {
+  id: 'number_format',
```

Comment:
Why do you need a default value for the ID. It should have no id by default unless it passed from the parent component.